source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -0,0 +1,139 @@
+#include "extensions/filters/http/aws_lambda/aws_lambda_filter.h"
+
+#include <string>
+#include <vector>
+
+#include "envoy/upstream/upstream.h"
+
+#include "common/common/fmt.h"
+#include "common/common/hex.h"
+#include "common/crypto/utility.h"
+#include "common/http/headers.h"
+#include "common/protobuf/utility.h"
+
+#include "extensions/filters/http/well_known_names.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace AwsLambdaFilter {
+
+namespace {
+
+constexpr auto filter_metadata_key = "com.amazonaws.lambda";
+constexpr auto egress_gateway_metadata_key = "egress_gateway";
+
+void setHeaders(Http::HeaderMap& headers, absl::string_view function_name) {
+  headers.setMethod(Http::Headers::get().MethodValues.Post);
+  headers.setPath(fmt::format("/2015-03-31/functions/{}/invocations", function_name));
```

Comment:
Would we have to have an updated release of this filter if "2015-03-31" changes in the future?   Could we make this a configurable item?

Reply:
Good question. I'm not inclined to make it configurable for a bunch of reasons:
1. AWS will always ensure this API works.
1. I want to reduce knobs.
1. It can be made configurable in the future.

Let me know your thoughts.